CRW-1233 remove duplicate/redundant/old related images

nickboldt opened the pull request:
CRW-1233 fix for redundant entries in csv: remove entries if already a digest (only insert if a tag being converted)

improve console logging so it's more obvious WHICH file is running; dedupe/sort/uniq the list of containers to collect; when checking registries for related images, check osbs if can't find rhec image

CRW-1237 temporarily disable switching to registry-proxy URLs - might have to re-enable these if osbs application broken

Change-Id: Iee5138612d378dba94951a63268ca104da39179b
Signed-off-by: nickboldt <nboldt@redhat.com>

Change-Id: I13374d23a3ac838d5f02a49d1cd3b937de18fa85
Signed-off-by: nickboldt <nboldt@redhat.com>

Change-Id: I978dabd7284a31f3d8bf53a40a8d8fc773fdd6dd
Signed-off-by: nickboldt <nboldt@redhat.com>

amisevsk commented on build/scripts/addDigests.sh:
```diff
@@ -16,18 +16,18 @@ QUIET=""
 PODMAN=$(command -v podman)
 if [[ ! -x $PODMAN ]]; then
-  echo "[WARNING] podman is not installed."
+  echo "[WARNING] ${0##*/} :: podman is not installed."
```
Alternative: `$(basename $0)`

nickboldt replied:
why use a built-in when a substring will do?

amisevsk commented on build/scripts/addDigests.sh:
```diff
@@ -54,32 +54,40 @@ rm -Rf ${BASE_DIR}/generated/${CSV_NAME}/
-if [[ ! ${CSV_FILE} ]]; then echo "[ERROR] Could not find CSV to generate in ${BASE_DIR}/generated/${CSV_NAME}/ !"; exit 1; fi
-echo "[INFO] CSV to generate: ${CSV_FILE}"
+# TODO: CRW-1044 support OCP 4.5 format until we switch to OCP 4.6
+if [[ -d $(find ${BASE_DIR}/generated/${CSV_NAME}/v${VERSION}/ || true) ]]; then
```
I'm not sure I understand the purpose of `|| true` here, but I think my bash is just lacking. Is this equivalent to 
```bash
if [[ -d ${BASE_DIR}/generated/${CSV_NAME}/v${VERSION}/ ]]; then
```
?

nickboldt replied:
to avoid the script pooping out if the dir doesn't exist?

amisevsk commented on build/scripts/addDigests.sh:
```diff
@@ -54,32 +54,40 @@ rm -Rf ${BASE_DIR}/generated/${CSV_NAME}/
+if [[ -d $(find ${BASE_DIR}/generated/${CSV_NAME}/v${VERSION}/ || true) ]]; then
+  CSV_FILE="$(find ${BASE_DIR}/generated/${CSV_NAME}/v${VERSION}/ -name "${CSV_NAME}.*${VERSION}.clusterserviceversion.yaml" -o -name "${CSV_NAME}.csv.yaml" | tail -1)"
+fi
+if [[ ! $CSV_FILE ]]; then 
```
`if [[ -z "$CSV_FILE" ]]; then` ?

amisevsk commented on build/scripts/addDigests.sh:
```diff
@@ -54,32 +54,40 @@ rm -Rf ${BASE_DIR}/generated/${CSV_NAME}/
+# if [[ ! "${QUIET}" ]]; then cat ${BASE_DIR}/generated/digests-mapping.txt; fi
 for mapping in $(cat ${BASE_DIR}/generated/digests-mapping.txt)
 do
   source=$(echo "${mapping}" | sed -e 's/\(.*\)=.*/\1/')
   dest=$(echo "${mapping}" | sed -e 's/.*=\(.*\)/\1/')
```
Could these be
```
source=${mapping##=*}
dest=${mapping%%*=}
```

nickboldt replied:
probably valid too, but I prefer regex to substringing (except when I don't) :D

amisevsk reviewed (commented):
godspeed

nickboldt commented:
For what it's worth, here's the new console output when you run `./build/scripts/sync-che-olm-to-crw-olm.sh`

```
$➔ ./build/scripts/sync-che-olm-to-crw-olm.sh -v 2.4.0 -p 2.3.0 -s ~/50/op/ -t ~/51/op --che 9.9.9-nightly.1598450052
[INFO] sync-che-olm-to-crw-olm.sh :: Converted (sed) /home/nboldt/51/op/manifests/codeready-workspaces.csv.yaml
[INFO] sync-che-olm-to-crw-olm.sh :: Converted (yq #1) /home/nboldt/51/op/manifests/codeready-workspaces.csv.yaml:
[INFO] sync-che-olm-to-crw-olm.sh ::  * .spec.displayName: "Red Hat CodeReady Workspaces"
[INFO] sync-che-olm-to-crw-olm.sh :: RELATED_IMAGE_keycloak_ppc64le = registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8:7.4
[INFO] sync-che-olm-to-crw-olm.sh :: RELATED_IMAGE_keycloak_s390x = registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8:7.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_machineexec = registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java11_openj9 = registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java11_s390x = registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java11_ppc64le = registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java11 = registry.redhat.io/codeready-workspaces/plugin-java11-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java8_openj9 = registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java8_s390x = registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java8_ppc64le = registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_java8 = registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_kubernetes = registry.redhat.io/codeready-workspaces/plugin-kubernetes-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_plugin_openshift = registry.redhat.io/codeready-workspaces/plugin-openshift-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_stacks_cpp = registry.redhat.io/codeready-workspaces/stacks-cpp-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_stacks_dotnet = registry.redhat.io/codeready-workspaces/stacks-dotnet-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_stacks_golang = registry.redhat.io/codeready-workspaces/stacks-golang-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_stacks_php = registry.redhat.io/codeready-workspaces/stacks-php-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_theia_endpoint = registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_codeready_workspaces_theia = registry.redhat.io/codeready-workspaces/theia-rhel8:2.4
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_jboss_eap_7_eap73_openjdk8_openshift_rhel7 = registry.redhat.io/jboss-eap-7/eap73-openjdk8-openshift-rhel7:7.3.1
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_jboss_eap_7_eap_xp1_openj9_11_openshift = registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_jboss_eap_7_eap_xp1_openjdk11_openshift_s390x = registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_jboss_eap_7_eap_xp1_openjdk11_openshift_ppc64le = registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_jboss_eap_7_eap_xp1_openjdk11_openshift = registry.redhat.io/jboss-eap-7/eap-xp1-openjdk11-openshift-rhel8:1.0-3
[INFO] insert-related-images-to-csv.sh :: RELATED_IMAGE_rhscl_mongodb_36_rhel7 = registry.redhat.io/rhscl/mongodb-36-rhel7:1-50
[INFO] sync-che-olm-to-crw-olm.sh :: Converted + inserted (yq #2) /home/nboldt/51/op/manifests/codeready-workspaces.csv.yaml:
 * CHE_FLAVOR: "codeready"
 * CHE_VERSION: "2.4.0"
 * CONSOLE_LINK_NAME: "che"
[INFO] addDigests.sh :: CSV to generate: /home/nboldt/51/op/generated/codeready-workspaces/codeready-workspaces.csv.yaml
[INFO] addDigests.sh :: CSV to update:   /home/nboldt/51/op/manifests/codeready-workspaces.csv.yaml
[INFO] buildDigestMap.sh :: Get images from CSV /home/nboldt/51/op/generated/codeready-workspaces/codeready-workspaces.csv.yaml
Trying to pull registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4...
Trying to pull registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4...
  error parsing HTTP 404 response body: invalid character 'N' looking for beginning of value: "Not found\n"
Error: unable to pull registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4: Error initializing source docker://registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4: Error reading manifest 2.4 in registry.redhat.io/codeready-workspaces/devfileregistry-rhel8: error parsing HTTP 404 response body: invalid character 'N' looking for beginning of value: "Not found\n"
Trying to pull registry-proxy.engineering.redhat.com/rh-osbs/codeready-workspaces-devfileregistry-rhel8:2.4...
Getting image source signatures
Copying blob sha256:c4d668e229cd131e0a8e4f8218dca628d9cf9697572875e355fe4b247b6aa9f0
Copying blob sha256:ec1681b6a383e4ecedbeddd5abc596f3de835aed6db39a735f62395c8edbff30
Copying blob sha256:da1cc572023a942fff15d59aefa5abbb59d2c24a03966db8074ef8f9bab277d4
Copying blob sha256:3a141aadcc99152a164669f691247b45e380680c83750b4e7807fa4d50d05329
Copying blob sha256:868e0ad21d932ac42b4a89574a309a314bc7a4d7a4f083791d3f704268f000cf
Copying config sha256:72ae2c43e7664d00f6d66818a8139003a554608ed91f6e702556d6bcc6ccb794
Writing manifest to image destination
Storing signatures
72ae2c43e7664d00f6d66818a8139003a554608ed91f6e702556d6bcc6ccb794
[INFO] buildDigestMap.sh :: Found 9 images in registry
Trying to pull registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4...
Trying to pull registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4...
  error parsing HTTP 404 response body: invalid character 'N' looking for beginning of value: "Not found\n"
Error: unable to pull registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4: Error initializing source docker://registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4: Error reading manifest 2.4 in registry.redhat.io/codeready-workspaces/pluginregistry-rhel8: error parsing HTTP 404 response body: invalid character 'N' looking for beginning of value: "Not found\n"
Trying to pull registry-proxy.engineering.redhat.com/rh-osbs/codeready-workspaces-pluginregistry-rhel8:2.4...
Getting image source signatures
Copying blob sha256:e9925e534450310c6fce98a0ae994963ccfb0b988573f420d13341621c1b6a5f
Copying blob sha256:c4d668e229cd131e0a8e4f8218dca628d9cf9697572875e355fe4b247b6aa9f0
Copying blob sha256:ec1681b6a383e4ecedbeddd5abc596f3de835aed6db39a735f62395c8edbff30
Copying blob sha256:da1cc572023a942fff15d59aefa5abbb59d2c24a03966db8074ef8f9bab277d4
Copying blob sha256:3a141aadcc99152a164669f691247b45e380680c83750b4e7807fa4d50d05329
Copying config sha256:5e2199d00a0af05bbbc86f8356f5d2f6f04fb06dbb111e5aefea20a9fb91714f
Writing manifest to image destination
Storing signatures
5e2199d00a0af05bbbc86f8356f5d2f6f04fb06dbb111e5aefea20a9fb91714f
[INFO] buildDigestMap.sh :: Found 11 images in registry
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/crw-2-rhel8-operator:2.4 (registry.redhat.io/codeready-workspaces/crw-2-rhel8-operator:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:89763ddec38a5925a052fa7ea75fc5a0db39124cada1e2d33b6eba3e32e8a7c6 # registry.redhat.io/codeready-workspaces/crw-2-rhel8-operator:2.4
[INFO] buildDigestMap.sh :: + Get digest for registry-proxy.engineering.redhat.com/rh-osbs/codeready-workspaces-devfileregistry-rhel8:2.4 (registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:7702adb0ed28b635e45804e87fe5dd98bdd3aa766fed7845a8ce509b91c22e36 # registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/jwtproxy-rhel8:2.4 (registry.redhat.io/codeready-workspaces/jwtproxy-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:8afecd5b0edc7734532ee76ff9eac1fc4814d8aaa6c9be440a2a88a20c014e4e # registry.redhat.io/codeready-workspaces/jwtproxy-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/machineexec-rhel8:2.4 (registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:c9bebc895e5fa5a0bd4ecaedfd5384ab75a45a96b6314ba5d4a6f4c1e8e109f9 # registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/pluginbroker-artifacts-rhel8:2.4 (registry.redhat.io/codeready-workspaces/pluginbroker-artifacts-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:d0eebf2c8b460adb75dc6bc5200aa9fd40d030b7b17c6b1c3b9d3c879f4652ee # registry.redhat.io/codeready-workspaces/pluginbroker-artifacts-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/pluginbroker-metadata-rhel8:2.4 (registry.redhat.io/codeready-workspaces/pluginbroker-metadata-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:cff23432d1d397bbbc7df65be9d6ddf4a97a3ef1801708bb7bb7d2fa72dbcce3 # registry.redhat.io/codeready-workspaces/pluginbroker-metadata-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-java11-openj9-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:27a71612f9bd3bee77adb4e164c44c61cf5085458d592215b2fe74c55d11abc6 # registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-java11-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-java11-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:e9deebbc320d28a2f425e858ed3dcf87fc67a40f6654d6eb7c2b6feea022b7d6 # registry.redhat.io/codeready-workspaces/plugin-java11-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-java8-openj9-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:14f2774e92b70d85280e506f81e2ea9a89c26490fd53a4421df8a694bd944d2d # registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-java8-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:d04f70c8340abaee1a282b77158d054f4faf2225bc17c79aafb413396c367782 # registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-kubernetes-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-kubernetes-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:d87aed64704369a50d1e54a57815b699f74d4efad1401d1a638808e655a37e48 # registry.redhat.io/codeready-workspaces/plugin-kubernetes-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/plugin-openshift-rhel8:2.4 (registry.redhat.io/codeready-workspaces/plugin-openshift-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:9c43a02b0dd0f66744359c5ccdb1f1780ecd92c3dc31b14d73b553ba763af8ab # registry.redhat.io/codeready-workspaces/plugin-openshift-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for registry-proxy.engineering.redhat.com/rh-osbs/codeready-workspaces-pluginregistry-rhel8:2.4 (registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:487941489c197f3165f6f901d22ac1d9d81079db5e048dc33a23beb5a2066814 # registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/server-rhel8:2.4 (registry.redhat.io/codeready-workspaces/server-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:63bf304cd04576048012693e7e8544a5a703790f99551554a75798bc799b112b # registry.redhat.io/codeready-workspaces/server-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/stacks-cpp-rhel8:2.4 (registry.redhat.io/codeready-workspaces/stacks-cpp-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:56543cfeeeac030821557ac4937db40f6845e874193c79c30267a680f9b2cbe7 # registry.redhat.io/codeready-workspaces/stacks-cpp-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/stacks-dotnet-rhel8:2.4 (registry.redhat.io/codeready-workspaces/stacks-dotnet-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:13628110b96de0e516ff2dfa29cdcaee64cd8f8978052c8160c294c332dba9f0 # registry.redhat.io/codeready-workspaces/stacks-dotnet-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/stacks-golang-rhel8:2.4 (registry.redhat.io/codeready-workspaces/stacks-golang-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:fef91718ccebc4cd9b89999f6b5df83bf3d60fce657f6f44eda092100549af2c # registry.redhat.io/codeready-workspaces/stacks-golang-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/stacks-php-rhel8:2.4 (registry.redhat.io/codeready-workspaces/stacks-php-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:b75f498954fbe858c74f80a89d132ba3560f40c0f697b0cd9550ed5663078ef6 # registry.redhat.io/codeready-workspaces/stacks-php-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/theia-endpoint-rhel8:2.4 (registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:942e1e6328169508e3fff8fd96c575d7a423339ced17dbf5813d61d1971adaef # registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.4
[INFO] buildDigestMap.sh :: + Get digest for quay.io/crw/theia-rhel8:2.4 (registry.redhat.io/codeready-workspaces/theia-rhel8:2.4)
[INFO] buildDigestMap.sh :: * Got digest     sha256:78edc9f75680cbe7f63774d9dfbbc505401486a73c8e420380e1d3078bdf9f2a # registry.redhat.io/codeready-workspaces/theia-rhel8:2.4
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/jboss-eap-7/eap73-openjdk8-openshift-rhel7:7.3.1
[INFO] buildDigestMap.sh :: = Got digest     sha256:24dea0cfc154a23c1aeb6b46ade182d0f981362f36b7e6fb9c7d8531ac639fe0 # registry.redhat.io/jboss-eap-7/eap73-openjdk8-openshift-rhel7:7.3.1
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12
[INFO] buildDigestMap.sh :: = Got digest     sha256:d6a7bdbf4726fe0e0e54c0bce9b2257bbd2a165c37cb4ec68e1f994716ffb15c # registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/jboss-eap-7/eap-xp1-openjdk11-openshift-rhel8:1.0-3
[INFO] buildDigestMap.sh :: = Got digest     sha256:94e1cd4eb4196a358e301c1992663258c0016c80247f507fd1c39cf9a73da833 # registry.redhat.io/jboss-eap-7/eap-xp1-openjdk11-openshift-rhel8:1.0-3
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/rhel8/postgresql-96:1
[INFO] buildDigestMap.sh :: = Got digest     sha256:fdc2398a25530547354714f2538c691d91b700e0cedef5361a3e7d96ddfd4e11 # registry.redhat.io/rhel8/postgresql-96:1
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/rhscl/mongodb-36-rhel7:1-50
[INFO] buildDigestMap.sh :: = Got digest     sha256:9f799d356d7d2e442bde9d401b720600fd9059a3d8eefea6f3b2ffa721c0dc73 # registry.redhat.io/rhscl/mongodb-36-rhel7:1-50
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8:7.4
[INFO] buildDigestMap.sh :: = Got digest     sha256:8e6c7874247053df431c25552c6e2edb050b2627ae21907149f419e0d9909135 # registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8:7.4
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/rh-sso-7/sso74-openshift-rhel8:7.4
[INFO] buildDigestMap.sh :: = Got digest     sha256:ec6801343eb1ca085154d8d7481552f2e9debc414125413d25e42216aa5922af # registry.redhat.io/rh-sso-7/sso74-openshift-rhel8:7.4
[INFO] buildDigestMap.sh :: - Get digest for registry.redhat.io/ubi8-minimal:8.2
[INFO] buildDigestMap.sh :: = Got digest     sha256:5cfbaf45ca96806917830c183e9f37df2e913b187aadb32e89fd83fa455ebaa6 # registry.redhat.io/ubi8-minimal:8.2
[INFO] addDigests.sh :: Map codeready-workspaces-crw-2-rhel8-operator = registry.redhat.io/codeready-workspaces/crw-2-rhel8-operator:2.4=registry.redhat.io/codeready-workspaces/crw-2-rhel8-operator@sha256:89763ddec38a5925a052fa7ea75fc5a0db39124cada1e2d33b6eba3e32e8a7c6
[INFO] addDigests.sh :: Map codeready-workspaces-devfileregistry-rhel8 = registry.redhat.io/codeready-workspaces/devfileregistry-rhel8:2.4=registry.redhat.io/codeready-workspaces/devfileregistry-rhel8@sha256:7702adb0ed28b635e45804e87fe5dd98bdd3aa766fed7845a8ce509b91c22e36
[INFO] addDigests.sh :: Map codeready-workspaces-jwtproxy-rhel8 = registry.redhat.io/codeready-workspaces/jwtproxy-rhel8:2.4=registry.redhat.io/codeready-workspaces/jwtproxy-rhel8@sha256:8afecd5b0edc7734532ee76ff9eac1fc4814d8aaa6c9be440a2a88a20c014e4e
[INFO] addDigests.sh :: Map codeready-workspaces-machineexec-rhel8 = registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.4=registry.redhat.io/codeready-workspaces/machineexec-rhel8@sha256:c9bebc895e5fa5a0bd4ecaedfd5384ab75a45a96b6314ba5d4a6f4c1e8e109f9
[INFO] addDigests.sh :: Map codeready-workspaces-pluginbroker-artifacts-rhel8 = registry.redhat.io/codeready-workspaces/pluginbroker-artifacts-rhel8:2.4=registry.redhat.io/codeready-workspaces/pluginbroker-artifacts-rhel8@sha256:d0eebf2c8b460adb75dc6bc5200aa9fd40d030b7b17c6b1c3b9d3c879f4652ee
[INFO] addDigests.sh :: Map codeready-workspaces-pluginbroker-metadata-rhel8 = registry.redhat.io/codeready-workspaces/pluginbroker-metadata-rhel8:2.4=registry.redhat.io/codeready-workspaces/pluginbroker-metadata-rhel8@sha256:cff23432d1d397bbbc7df65be9d6ddf4a97a3ef1801708bb7bb7d2fa72dbcce3
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-java11-openj9-rhel8 = registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-java11-openj9-rhel8@sha256:27a71612f9bd3bee77adb4e164c44c61cf5085458d592215b2fe74c55d11abc6
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-java11-rhel8 = registry.redhat.io/codeready-workspaces/plugin-java11-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-java11-rhel8@sha256:e9deebbc320d28a2f425e858ed3dcf87fc67a40f6654d6eb7c2b6feea022b7d6
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-java8-openj9-rhel8 = registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-java8-openj9-rhel8@sha256:14f2774e92b70d85280e506f81e2ea9a89c26490fd53a4421df8a694bd944d2d
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-java8-rhel8 = registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-java8-rhel8@sha256:d04f70c8340abaee1a282b77158d054f4faf2225bc17c79aafb413396c367782
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-kubernetes-rhel8 = registry.redhat.io/codeready-workspaces/plugin-kubernetes-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-kubernetes-rhel8@sha256:d87aed64704369a50d1e54a57815b699f74d4efad1401d1a638808e655a37e48
[INFO] addDigests.sh :: Map codeready-workspaces-plugin-openshift-rhel8 = registry.redhat.io/codeready-workspaces/plugin-openshift-rhel8:2.4=registry.redhat.io/codeready-workspaces/plugin-openshift-rhel8@sha256:9c43a02b0dd0f66744359c5ccdb1f1780ecd92c3dc31b14d73b553ba763af8ab
[INFO] addDigests.sh :: Map codeready-workspaces-pluginregistry-rhel8 = registry.redhat.io/codeready-workspaces/pluginregistry-rhel8:2.4=registry.redhat.io/codeready-workspaces/pluginregistry-rhel8@sha256:487941489c197f3165f6f901d22ac1d9d81079db5e048dc33a23beb5a2066814
[INFO] addDigests.sh :: Map codeready-workspaces-server-rhel8 = registry.redhat.io/codeready-workspaces/server-rhel8:2.4=registry.redhat.io/codeready-workspaces/server-rhel8@sha256:63bf304cd04576048012693e7e8544a5a703790f99551554a75798bc799b112b
[INFO] addDigests.sh :: Map codeready-workspaces-stacks-cpp-rhel8 = registry.redhat.io/codeready-workspaces/stacks-cpp-rhel8:2.4=registry.redhat.io/codeready-workspaces/stacks-cpp-rhel8@sha256:56543cfeeeac030821557ac4937db40f6845e874193c79c30267a680f9b2cbe7
[INFO] addDigests.sh :: Map codeready-workspaces-stacks-dotnet-rhel8 = registry.redhat.io/codeready-workspaces/stacks-dotnet-rhel8:2.4=registry.redhat.io/codeready-workspaces/stacks-dotnet-rhel8@sha256:13628110b96de0e516ff2dfa29cdcaee64cd8f8978052c8160c294c332dba9f0
[INFO] addDigests.sh :: Map codeready-workspaces-stacks-golang-rhel8 = registry.redhat.io/codeready-workspaces/stacks-golang-rhel8:2.4=registry.redhat.io/codeready-workspaces/stacks-golang-rhel8@sha256:fef91718ccebc4cd9b89999f6b5df83bf3d60fce657f6f44eda092100549af2c
[INFO] addDigests.sh :: Map codeready-workspaces-stacks-php-rhel8 = registry.redhat.io/codeready-workspaces/stacks-php-rhel8:2.4=registry.redhat.io/codeready-workspaces/stacks-php-rhel8@sha256:b75f498954fbe858c74f80a89d132ba3560f40c0f697b0cd9550ed5663078ef6
[INFO] addDigests.sh :: Map codeready-workspaces-theia-endpoint-rhel8 = registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.4=registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8@sha256:942e1e6328169508e3fff8fd96c575d7a423339ced17dbf5813d61d1971adaef
[INFO] addDigests.sh :: Map codeready-workspaces-theia-rhel8 = registry.redhat.io/codeready-workspaces/theia-rhel8:2.4=registry.redhat.io/codeready-workspaces/theia-rhel8@sha256:78edc9f75680cbe7f63774d9dfbbc505401486a73c8e420380e1d3078bdf9f2a
[INFO] addDigests.sh :: Map jboss-eap-7-eap73-openjdk8-openshift-rhel7 = registry.redhat.io/jboss-eap-7/eap73-openjdk8-openshift-rhel7:7.3.1=registry.redhat.io/jboss-eap-7/eap73-openjdk8-openshift-rhel7@sha256:24dea0cfc154a23c1aeb6b46ade182d0f981362f36b7e6fb9c7d8531ac639fe0
[INFO] addDigests.sh :: Map jboss-eap-7-eap-xp1-openj9-11-openshift-rhel8 = registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8:1.0-12=registry.redhat.io/jboss-eap-7/eap-xp1-openj9-11-openshift-rhel8@sha256:d6a7bdbf4726fe0e0e54c0bce9b2257bbd2a165c37cb4ec68e1f994716ffb15c
[INFO] addDigests.sh :: Map jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8 = registry.redhat.io/jboss-eap-7/eap-xp1-openjdk11-openshift-rhel8:1.0-3=registry.redhat.io/jboss-eap-7/eap-xp1-openjdk11-openshift-rhel8@sha256:94e1cd4eb4196a358e301c1992663258c0016c80247f507fd1c39cf9a73da833
[INFO] addDigests.sh :: Map rhel8-postgresql-96 = registry.redhat.io/rhel8/postgresql-96:1=registry.redhat.io/rhel8/postgresql-96@sha256:fdc2398a25530547354714f2538c691d91b700e0cedef5361a3e7d96ddfd4e11
[INFO] addDigests.sh :: Map rhscl-mongodb-36-rhel7 = registry.redhat.io/rhscl/mongodb-36-rhel7:1-50=registry.redhat.io/rhscl/mongodb-36-rhel7@sha256:9f799d356d7d2e442bde9d401b720600fd9059a3d8eefea6f3b2ffa721c0dc73
[INFO] addDigests.sh :: Map rh-sso-7-sso74-openj9-openshift-rhel8 = registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8:7.4=registry.redhat.io/rh-sso-7/sso74-openj9-openshift-rhel8@sha256:8e6c7874247053df431c25552c6e2edb050b2627ae21907149f419e0d9909135
[INFO] addDigests.sh :: Map rh-sso-7-sso74-openshift-rhel8 = registry.redhat.io/rh-sso-7/sso74-openshift-rhel8:7.4=registry.redhat.io/rh-sso-7/sso74-openshift-rhel8@sha256:ec6801343eb1ca085154d8d7481552f2e9debc414125413d25e42216aa5922af
[INFO] addDigests.sh :: Map ubi8-minimal:8.2=registry.redhat.io-ubi8-minimal = registry.redhat.io/ubi8-minimal:8.2=registry.redhat.io/ubi8-minimal@sha256:5cfbaf45ca96806917830c183e9f37df2e913b187aadb32e89fd83fa455ebaa6
[INFO] addDigests.sh :: CSV updated: manifests/codeready-workspaces.csv.yaml
```
